packages/grid/x-data-grid-pro/src/hooks/features/lazyLoader/useGridLazyLoader.ts

```diff
@@ -0,0 +1,233 @@
+import * as React from 'react';
+import {
+  useGridApiEventHandler,
+  GridFeatureModeConstant,
+  GridRenderedRowsIntervalChangeParams,
+  useGridSelector,
+  GridRowId,
+  gridVisibleSortedRowIdsSelector,
+  gridSortModelSelector,
+  gridFilterModelSelector,
+  useGridApiOptionHandler,
+  GridEventListener,
+  GridRowEntry,
+  GridDimensions,
+  GridFeatureMode,
+} from '@mui/x-data-grid';
+import { useGridVisibleRows } from '@mui/x-data-grid/internals';
+import { getRenderableIndexes } from '@mui/x-data-grid/hooks/features/virtualization/useGridVirtualScroller';
+import { GridApiPro } from '../../../models/gridApiPro';
+import {
+  DataGridProProcessedProps,
+  GridExperimentalProFeatures,
+} from '../../../models/dataGridProProps';
+import { GRID_SKELETON_ROW_ROOT_ID } from './useGridLazyLoaderPreProcessors';
+import { GridFetchRowsParams } from '../../../models/gridFetchRowsParams';
+
+function findSkeletonRowsSection(
+  visibleRows: GridRowEntry<any>[],
+  range: { firstRowIndex: number; lastRowIndex: number },
+): { firstRowIndex: number; lastRowIndex: number } {
```

Comment:
```suggestion
) {
```

to infer the return type